src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -955,7 +983,12 @@ impl LinkCollector<'_, '_> {
         let link = ori_link.replace("`", "");
         let parts = link.split('#').collect::<Vec<_>>();
         let (link, extra_fragment) = if parts.len() > 2 {
-            anchor_failure(cx, &item, &link, dox, link_range, AnchorFailure::MultipleAnchors);
+            // A valid link can't have multiple #'s
+            anchor_failure(
+                cx,
+                DiagnosticInfo { item, dox, path_str: &link, link_range },
```

Comment:
nit: might be easier to construct `DiagnosticInfo` right away so you can use `DiagnosticInfo { path_str: &link, ..diag_info }` here.